qdrant-landing/content/articles/binary-quantization.md

```diff
@@ -0,0 +1,236 @@
+
+**With binary quantization, those same 100K OpenAI vectors only require 128 MB of RAM.** We benchmarked this result using methods similar to those covered in our [Scalar Quantization memory estimation](/articles/scalar-quantization/#benchmarks).
+
+This reduction in RAM needed is achieved through the compression that happens in the binary conversion. Instead of putting the HNSW index for the full vectors into RAM, we just put the binary vectors into RAM, use them for the initial oversampled search, and then use the HNSW full index of the oversampled results for the final precise search. All of this happens under the hoods without any intervention needed on your part. 
+
+#### When should you not use BQ?
+
+Since this method exploits the over-parameterization of embedding, you can expect poorer results for small embeddings i.e. less than 1024 dimensions. With the smaller number of elements, there is not enough information maintained in the binary vector to achieve good results. 
+
+You will still get faster boolean operations and reduced RAM usage, but the accuracy degradation might be too high. 
+
+## Sample Implementation
+
+Now that we have introduced you to binary quantization, let’s try our a basic implementation. In this example, we will be using OpenAI and Cohere with Qdrant.
+
+#### Create a collection with Binary Quantization enabled
+
+Here is what you should do at indexing time when you create the collection: 
+
+1. We store all the "full" vectors on disk.
+2. Then we set the binary embeddings to be in RAM.
+
+By default, both the full vectors and BQ get stored in RAM. We move the full vectors to disk because this saves us memory and allows us to store more vectors in RAM. By doing this, we explicitly move the binary vectors to memory by setting `always_ram=True`. 
+
+```python
+from qdrant_client import QdrantClient
+
+#collect to our Qdrant Server
+client = QdrantClient(
+    url="http://localhost:6334",
+    timeout=600,
+    prefer_grpc=True,
+)
+
+#Create the collection to hold our embeddings
+# on_disk=True and the quantization_config are the areas to focus on
+collection_name = "binary-quantization"
+client.recreate_collection(
+    collection_name=f"{collection_name}",
+    vectors_config=models.VectorParams(
+        size=1536,
+        distance=models.Distance.DOT,
+        on_disk=True,
+    ),
+    optimizers_config=models.OptimizersConfigDiff(
+        default_segment_number=5,
+        indexing_threshold=0,
+    ),
+    quantization_config=models.BinaryQuantization(
+        binary=models.BinaryQuantizationConfig(always_ram=True),
+    ),
+)
+```
+
+#### What is happening in the OptimizerConfig? 
+
+We're setting `indexing_threshold` to 0 i.e. disabling the indexing to zero. This allows faster uploads of vectors and payloads. We will turn it back on down below, once all the data is loaded
```

Comment:
Are there some reasons for bulk upload here? I propose to use a small simple config. Bulk upload is a separate topic